src/api.js

```diff
@@ -833,6 +853,7 @@ API.prototype.router = function(options) {
 API.prototype.reference = function(options) {
   assert(options,         "Options is required");
   assert(options.baseUrl, "A 'baseUrl' must be provided");
+  this._entries.push(ping);
```

Comment:
It looks like this is getting pushed twice now.  I don't think `this._entries` should be modified by functions that don't say they do so (for example, it's reasonable to think I could call `api.reference()` twice and get the same answer both times).

In both of these cases, could you do something like `_.concat(this._entries, [ping]).forEach`.  That joins the two arrays but doesn't modify `this._entries` in the process.